### PR TITLE
[Impl] BTC Endpoints

### LINKED ISSUE #61: [Feature] finish adding bitcoin endpoints
still need:
- [x] /construction/metadata
- [x] /construction/parse
- [x] /construction/payloads
- [x] /construction/preprocess

### PR DESCRIPTION
**Pull request by jels-boolean**
Fixes #61

### CODE REVIEW

**Comment by jels-boolean** [comment]
Time taken: 7 hours

**Comment by MACS-J1149** [comment]
going to work on testing these today/over the weekend now that i have some free time

File: rosetta-bitcoin/src/construction_api.rs [hunk 1]
```diff
@@ -78,41 +103,216 @@ impl ConstructionApi for BitcoinConstructionApi {
...
+            input: vec![],
+            output: vec![],
+        };
+
+        let coins = data
+            .metadata
+            .get("coins")
+            .ok_or_else(|| MentatError::from("no coins provided"))?
+            .as_array()
+            .ok_or_else(|| MentatError::from("malformed coins field in metadata"))?;
```
[the remark this answers is not in the document]
Comment (MACS-J1149) [reply rosetta-bitcoin/src/construction_api.rs]: talked further with @jels-boolean and it looks like this is how its done in the coinbase spec. im just going to try and get some examples working on their end so i can compare them against ours and make sure it all matches

**Comment by gluax** [comment]
Going to merge this because the main branch is not being updated to use the new generic RpcCaller methods is blocking me from implementing the health check.

We can test it later and file bugs.

**Comment by mowkoshka** [comment]
Going to merge this because the main branch is not being updated to use the new generic RpcCaller methods is blocking me from implementing the health check.

We can test it later and file bugs.

**Pull request merged**